numba/tests/test_parfors.py

```diff
@@ -44,6 +46,83 @@
 import cmath
 import unittest
 
+# NOTE: Each parfors test class is run in separate subprocess, this to reduce
+# memory pressure in CI settings. The environment variable "SUBPROC_TEST" is
+# used to determine whether a test is skipped or not, such that if you want to
+# run any parfors test directly this environment variable can be set. The
+# subprocesses running the test classes set this environment variable as the new
+# process starts which enables the tests within the process.
+_exec_cond = os.environ.get('SUBPROC_TEST', None) == '1'
+needs_subprocess = unittest.skipUnless(_exec_cond, "needs subprocess harness")
+
+
+@skip_parfors_unsupported
+class TestParforsRunner(TestCase):
+
+    _numba_parallel_test_ = False
+
+    # Each test class can run for 30 minutes before time out.
+    _TIMEOUT = 1800
+
+    """This is the test runner for all the parfors tests, it runs them in
+    subprocesses as described above. The convention for the test method naming
+    is: `test_<TestClass>` where <TestClass> is the name of the test class in
+    this module.
+    """
+    def runner(self):
+        themod = self.__module__
+        test_clazz_name = self.id().split('.')[-1].split('_')[-1]
+        the_test = f'{themod}.{test_clazz_name}'
+        cmd = [sys.executable, '-m', 'numba.runtests', the_test]
+        env_copy = os.environ.copy()
+        env_copy['SUBPROC_TEST'] = '1'
+        status = subprocess.run(cmd, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE, timeout=self._TIMEOUT,
+                                env=env_copy, universal_newlines=True)
+        self.assertEqual(status.returncode, 0)
```

Comment:
```suggestion
        self.assertEqual(status.returncode, 0, msg=status.stderr)
```
so that we know what tests are failing.